Document the publishing and scheduling workflow

CosmeValera opened the pull request:
Turns the notes from @CosmeValera into `docs/publishing.md`, linked from the README.

## Why it changed shape

**It was in the wrong place.** Zola renders everything under `content/`, so `content/README.md` was not a private note: it built as `/readme/` and landed in the sitemap and the search index.

**It documented a feature that does not exist.** The original text described a template snippet swapping a future post's date for today's. No such code is in `templates/`, and `.github/workflows/deploy.yml` only runs on push to `main` and manual dispatch, so no date can trigger a publish on its own.

## What the doc says now

- The filename date prefix **is** the page date: `2026-07-25-my-post.md` publishes as `/blog/my-post/`. No post in the repo sets `date` in its front matter.
- `draft = true` keeps a post out of the build entirely: no page, no sitemap entry, no search hit. `zola serve --drafts` to preview.
- To land a post on a chosen day, flip `draft = false` that morning and push. The deploy takes about a minute.
- What real date-based publishing would actually take, a `schedule:` trigger plus filtering future posts out of the templates **and** of `scripts/enrich-search-index.py`, `enrich-sitemap.py`, `generate-feed-json.py` and `generate-llms-txt.py`, is written down as the alternative it is. Half of it is the trap: the post vanishes from the index but keeps its URL and its sitemap entry.
- A note on why contributor docs go in `docs/` and not `content/`, so the original mistake does not repeat.

`main` is merged in, so the diff is just the two files. Verified with `zola build`: no `/readme/` page, no sitemap entry.

Still a draft on purpose, not for merge yet.

CosmeValera committed:
Add README.md for Zola Blog Scheduling in content. Specific implementation is missing

CosmeValera committed:
It needs this '---' at the beggining because otherwise zola doesn't compile haha

CosmeValera committed:
It needs this '---' at the beggining because otherwise zola doesn't compile haha

CosmeValera commented:
**I did not add the implementation. Here's why:**

So the thing is that in my code this is the implementation that I have in `blog.html`:
```html
{% extends "base.html" %}

{% block title %}
  Blog | Cosme
{% endblock %}

{% block styles %}
  <link rel="stylesheet" href="{{ get_url(path="blog.css") | safe }}">
{% endblock %}

{% block content %}
<div class="column right">
  <h1 class="page-title">Blog</h1>
  
  <div class="blog-cards">
    {% for post in section.pages %}
      {% set post_date_numeric = post.date | date(format="%Y%m%d") | int %}
      {% set now_numeric = now() | date(format="%Y%m%d") | int %}
      
      {% if post_date_numeric > now_numeric %}
        {% set post_date_display = now() | date(format="%B %d, %Y") %}
      {% else %}
        {% set post_date_display = post.date | date(format="%B %d, %Y") %}
      {% endif %}
      
      <a class="blog-card" href="{{ post.permalink }}">
        <h3 class="blog-title">{{ post.title }}</h3>
        <div class="blog-card-details">
          <p class="card-description"> {{ post.description }} </p>
          <span class="card-date"> {{ post_date_display }}</span>
        </div>
      </a>
    {% endfor %}
  </div>
</div>
{% endblock content %}
```

This is the whole file, and as you can see the implementation is pretty straighforward. I handle the date only in blog.html.

I tried applying the same in your `blog/index.html`, but it is not as straighforward. There are several files that depend on date somehow: `blog/index.html`, `blog/blog-post.html`, `helpers-preview.html`

For example, in `blog/index.html` you have `pinned_posts` and `normal_posts`. I'm not so sure how to deal with those in regard to implementing this functionaility.

If you find it interesting, feel free to add the implementation into this branch 🥰

Chemaclass reviewed (commented):
Thanks for bringing this idea to life! I will take it from here and work on it at least for the blog post section - which is the area where I think it will be the most interesting for my use cases. In the meanwhile I will mark the PR as draft

Chemaclass committed:
Merge branch 'main' into feature/zola-blog-scheduling

Chemaclass committed:
docs: move the publishing notes out of content/ and make them accurate

Zola renders everything under content/, so content/README.md was not a private
note: it built as /readme/ and landed in the sitemap and the search index. The
file moves to docs/publishing.md, where Zola never looks.

The text also described a feature that does not exist. It documented a template
snippet swapping a future post's date for today's, but no such code is in
templates/, and the deploy workflow only runs on push to main and manual
dispatch, so no date can trigger a publish on its own.

Document what the site actually does instead: the filename date prefix is the
page date (no post here sets `date` in front matter), `draft = true` keeps a post
out of the build entirely, and the way to land a post on a chosen day is to flip
the draft flag that morning. What real date-based publishing would take, a
scheduled workflow plus filtering in the templates and the post-build scripts, is
written down as the alternative it is.

Original notes by @CosmeValera in #16.

Chemaclass committed:
Merge branch 'main' into feature/zola-blog-scheduling
